CMSIS/NN/Include/arm_nnfunctions.h

```diff
@@ -189,35 +190,72 @@ extern    "C"
+   *
+   * @return         The function returns  required buffer size(bytes)
+   *
+   */
+    int32_t arm_convolve_wrapper_s8_get_buffer_size(const cmsis_nn_conv_params* conv_params,
+                                                    const cmsis_nn_dims* input_dims,
+                                                    const cmsis_nn_dims* filter_dims,
+                                                    const cmsis_nn_dims* output_dims);
+
+  /**
    * @brief Basic s8 convolution function
-   * @param[in]       input           pointer to input tensor. Range: int8, format: [N,H,W,in_ch]
-   * @param[in]       input_x         input tensor width
-   * @param[in]       input_y         input tensor height
-   * @param[in]       input_ch        number of input tensor channels
-   * @param[in]       input_batches   number of input batches
-   * @param[in]       kernel          pointer to kernel weights. Range: int8, format: [out_ch, H, W, in_ch]
-   * @param[in]       output_ch       number of filters, i.e., output tensor channels
-   * @param[in]       kernel_x        filter/kernel width
-   * @param[in]       kernel_y        filter/kernel height
-   * @param[in]       pad_x           padding along width
-   * @param[in]       pad_y           padding along height
-   * @param[in]       stride_x        convolution stride x
-   * @param[in]       stride_y        convolution stride y
-   * @param[in]       bias            pointer to per output channel bias. Range: int32
-   * @param[in,out]   output          pointer to output tensor. format: [H, W, out_ch]
-   * @param[in]       output_shift    pointer to per output channel requantization shift parameter.
-   * @param[in]       output_mult     pointer to per output channel requantization multiplier parameter.
-   * @param[in]       out_offset      output tensor offset. Range: int8
-   * @param[in]       input_offset    input tensor offset. Range: int8
-   * @param[in]       output_activation_min   Minimum value to clamp the output to. Range: int8
-   * @param[in]       output_activation_max   Minimum value to clamp the output to. Range: int8
-   * @param[in]       output_x    output tensor width
-   * @param[in]       output_y    output tensor height
-   * @param[in]       buffer_a    pointer to buffer space used for input optimization(partial im2col) and is necessary
-   *                              when ARM_MATH_DSP is defined.
-   *                              Required space: (2 * input_ch * kernel_x * kernel_y) * sizeof(q15_t) bytes
-   *                              Use arm_convolve_s8_get_buffer_size() to get the size.
+   * @param[in, out] ctx            Function context that contains the additional buffer if required by the implementation. 
+                                    arm_convolve_s8_get_buffer_size will return the buffer_size if required
+   * @param[in]      conv_params    Convolution parameters (e.g. strides, dilations, pads,...). Input and output offset range: [-127,128)
```

Comment:
input offset has a range of [-127, 128] but the range of output offset is [-128, 127] or int8